### PR TITLE
Add libm library when compiling

### PR DESCRIPTION
**Pull request by dramforever**
This makes it possible to use C standard math functions in RISC-V kata, which is pretty much indispensible for anything involving floating point.

I also refactored a little bit because the compile command was getting long.

### CODE REVIEW

**Comment by dramforever** [comment]
Probably should add similar option for NASM as well, though I can't find where the runner is. Haven't looked closely though, to be honest.

File: bin/run [hunk 1]
```diff
@@ -11,11 +11,15 @@ fi
 
 W=/workspace
 
-ARCH=rv64gcv_zba_zbb_zbc_zbs_zbkx_zk_zks
 QEMU_CPU=rv64,v=on,vext_spec=v1.0,zbkx=on,zk=on,zks=on
+ARCH=rv64gcv_zba_zbb_zbc_zbs_zbkx_zk_zks
+
+LIBS="-lcgreen -lm"
+FILES="solutions.s solution_tests.c codewars_reporter.c tests.c"
```
Comment (kazk) [inline bin/run]:
```suggestion
FILES="solution.s solution_tests.c codewars_reporter.c tests.c"
```

**Comment by kazk** [comment]
Failing with:

```text
/usr/bin/ld: /tmp/cc9kA2PF.o: in function `main':
tests.c:(.text.startup+0x1a): undefined reference to `run_test_suite'
/usr/bin/ld: /tmp/cc58oW6N.o: in function `.L0 ':
solution_tests.c:(.text+0x38): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0x52): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0x6a): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0x84): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0x9c): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0xb6): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0xd0): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0xea): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0x104): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0x11e): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0x138): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0x158): undefined reference to `assert_core_'
/usr/bin/ld: /tmp/cc58oW6N.o: in function `Multiply__works_for_100_random_tests':
solution_tests.c:(.text+0x1bc): undefined reference to `create_equal_to_value_constraint'
/usr/bin/ld: solution_tests.c:(.text+0x1ca): undefined reference to `assert_core_'
/usr/bin/ld: solution_tests.c:(.text+0x202): undefined reference to `create_named_test_suite_'
/usr/bin/ld: solution_tests.c:(.text+0x218): undefined reference to `add_test_'
/usr/bin/ld: solution_tests.c:(.text+0x22e): undefined reference to `add_test_'
/usr/bin/ld: /tmp/ccRiZgBF.o: in function `.L2':
codewars_reporter.c:(.text+0x3e): undefined reference to `reporter_finish_test'
/usr/bin/ld: /tmp/ccRiZgBF.o: in function `.L0 ':
codewars_reporter.c:(.text+0x270): undefined reference to `reporter_start_test'
/usr/bin/ld: /tmp/ccRiZgBF.o: in function `codewars_reporter_start_test':
codewars_reporter.c:(.text+0x2c0): undefined reference to `reporter_start_suite'
/usr/bin/ld: /tmp/ccRiZgBF.o: in function `codewars_reporter_start_suite':
codewars_reporter.c:(.text+0x31a): undefined reference to `reporter_finish_suite'
/usr/bin/ld: codewars_reporter.c:(.text+0x350): undefined reference to `create_reporter'
collect2: error: ld returned 1 exit status
```
https://github.com/codewars/riscv/pull/10/checks#step:6:6

**Comment by dramforever** [comment]
I'll go check it out, could be a library ordering problem

**Comment by kazk** [comment]
I'll let you know when it's deployed.

> Probably should add similar option for NASM as well

It does already. `nasm -f elf64 -o solution.o solution.nasm` followed by `gcc -std=c11 fixture.c solution.o criterion-hooks.o -o test -lcriterion -lm -lcrypto -ldl`. Reported and confirmed in https://github.com/codewars/codewars-runner-cli/issues/621

**Comment by kazk** [comment]
Deployed

**Pull request merged**